agent/engine/engine_unix_integ_test.go

```diff
@@ -1331,3 +1369,40 @@ func TestTaskLevelVolume(t *testing.T) {
 	client := taskEngine.(*DockerTaskEngine).client
 	client.RemoveVolume(context.TODO(), "TestTaskLevelVolume", 5*time.Second)
 }
+
+func TestGPUAssociationTask(t *testing.T) {
+	gpuSupportEnabled := utils.ParseBool(os.Getenv("ECS_ENABLE_GPU_SUPPORT"), false)
```

Comment:
nit: I'd rather use a specific env var for testing instead of reusing this one. The reason is that this could convey that this test will get run whenever agent is configured for GPU support.

Reply:
Fixed this problem by adding a check if the testing instance is p2 or p3 and if the /var/lib/ecs/ecs.config file has ECS_ENABLE_GPU_SUPPORT is set to true or not.